src/main/java/org/owasp/html/ExtensibleHtmlStreamRenderer.java

```diff
@@ -0,0 +1,501 @@
+            return;
+        }
+        if (pendingUnescaped != null) {
+            error("Tag content cannot appear inside CDATA element", elementName);
+            return;
+        }
+
+        escapingMode = HtmlTextEscapingMode.getModeForTag(elementName);
+
+        switch (escapingMode) {
+        case CDATA_SOMETIMES:
+        case CDATA:
+        case PLAIN_TEXT:
+            lastTagOpened = elementName;
+            pendingUnescaped = new StringBuilder();
+            break;
+        default:
+            break;
+        }
+
+        output.append('<').append(elementName);
+
+        for (Iterator<? extends String> attrIt = attrs.iterator(); attrIt.hasNext();) {
+            String name = attrIt.next();
+            String value = attrIt.next();
+            name = HtmlLexer.canonicalAttributeName(name);
+            if (!isValidHtmlName(name)) {
+                error("Invalid attr name", name);
+                continue;
+            }
+            output.append(' ').append(name).append('=').append('"');
+            Encoding.encodeHtmlAttribOnto(value, output);
+            if (value.indexOf('`') != -1) {
+                // Apparently, in quirks mode, IE8 does a poor job producing innerHTML
+                // values. Given
+                // <div attr="``foo=bar">
+                // we encode &#96; but if JavaScript does:
+                // nodeA.innerHTML = nodeB.innerHTML;
+                // and nodeB contains the DIV above, then IE8 will produce
+                // <div attr=``foo=bar>
+                // as the value of nodeB.innerHTML and assign it to nodeA.
+                // IE8's HTML parser treats `` as a blank attribute value and foo=bar
+                // becomes a separate attribute.
+                // Adding a space at the end of the attribute prevents this by forcing
+                // IE8 to put double quotes around the attribute when computing
+                // nodeB.innerHTML.
+                output.append(' ');
+            }
+            output.append('"');
+        }
+
+        // Limit our output to the intersection of valid XML and valid HTML5 when
+        // the output contains no special HTML5 elements like <title>, <script>, or
+        // <textarea>.
+        if (HtmlTextEscapingMode.isVoidElement(elementName)) {
+            output.append(" /");
+        }
+
+        output.append('>');
+    }
+
+    public final void closeTag(String elementName) {
+        try {
+            writeCloseTag(safeName(elementName));
+        } catch (IOException ex) {
+            ioExHandler.handle(ex);
+        }
+    }
+
+    protected void writeCloseTag(String uncanonElementName) throws IOException {
+        if (!open) {
+            throw new IllegalStateException();
+        }
+        String elementName = HtmlLexer.canonicalElementName(uncanonElementName);
+        if (!isValidHtmlName(elementName)) {
+            error("Invalid element name", elementName);
+            return;
+        }
+
```

Comment:
As I said earlier, please highlight the below change with a comment with respective flags meaning

Reply:
done